src/screens/WasteCollectionScreen.js

```diff
@@ -185,7 +155,7 @@ export const WasteCollectionScreen = ({ navigation }) => {
 
     setSelectedTypes(
       Object.fromEntries(
-        (waste.selectedTypeKeys?.length ? waste.selectedTypeKeys : Object.keys(usedTypes)).map(
+        (waste.selectedTypeKeys ? waste.selectedTypeKeys : Object.keys(usedTypes)).map(
```

Comment:
I think `selectedTypeKeys` is not a boolean expression, so I suggest using the `!!` control.

Reply:
As a result it is not a boolean expression. Here we are checking true/false. I think the !! check is necessary here, as we always do.

Reply:
we are not checking true/false. we are checking if defined or undefined. where do you think is the boolean condition placed or needed?

Reply:
ok i guess, if `selectedTypeKeys` is an array, we can use it. otherwise it falls back to `Object.keys(usedTypes)`. seems legit.